README.md

```diff
@@ -12,36 +12,31 @@ This SDK is still considered work in progress.
 
 **For v1 of the SDK, see** [CloudEvents Go SDK v1](./README_v1.md).
 
-**v2.0.0-preview2:**
+**v2.0.0-preview8:**
 
-In _preview2_ we are focusing on the new Client interface:
+In _preview8_ we are focusing on the new Client interface:
 
 ```go
 type Client interface {
-	Send(ctx context.Context, event event.Event) error
-	Request(ctx context.Context, event event.Event) (*event.Event, error)
+	Send(ctx context.Context, event event.Event) protcol.Result
+	Request(ctx context.Context, event event.Event) (*event.Event, protcol.Result)
 	StartReceiver(ctx context.Context, fn interface{}) error
 }
 ```
 
-Where a full `fn` looks like
-`func(context.Context, event.Event) (*event.Event, transport.Result)`
+`Send` and `Request` will return the result of the outbound event. This at minimum means the result is testable
+for being an _ACK_ or _NACK_ via:
 
-For protocols that do not support responses, `StartReceiver` will throw an error
-when attempting to set a receiver fn with that capability.
-
-For protocols that do not support responses from send (Requester interface),
-`Client.Request` will throw an error.
-
-**v2.0.0-preview1:**
-
-In _preview1_ we are focusing on the new interfaces found in pkg/transport (will
-be renamed to protocol):
+```go
+if cloudevents.IsACK(result) { 
+	// handle result as an accepted event.
+} else if cloudevents.IsNACK(result) {
+	// handle result as a rejected event.
+} else {
```

Comment:
if you tried to send bad data it will not NAC nor ACK, it will throw an error

I think the last else should be `else if result != nil {`

I will fix

Reply:
ah ok fine, can you add a comment about it?